ext/node/polyfills/internal/webstreams/adapters.js

```diff
@@ -526,10 +544,24 @@ export function newReadableStreamFromStreamReadable(
     // This is a protection against non-standard, legacy streams
     // that happen to emit an error event again after finished is called.
     streamReadable.on("error", () => {});
-    if (error) {
-      return controller.error(error);
+    
+    // Controller operations can fail if the stream is destroyed while
+    // processing the finish event. These errors are expected and safely ignored.
+    try {
+      if (error) {
+        controller.error(error);
+      } else {
+        controller.close();
+      }
+    } catch (controllerError) {
+      // Ignore errors from controller operations when the controller
+      // is already closed, cancelled, or errored. This can happen
+      // when the stream is destroyed during data emission.
+      debug(
+        `Ignoring controller.${error ? "error()" : "close()"} error (stream destroyed during data emission):`,
+        controllerError
+      );
     }
-    controller.close();
```

Comment:
Node.js keeps track of `wasCanceled` variable here - and if it was already cancelled, then `controller.close()` is never called.